backend/controllers/locationZoneController.js

```diff
@@ -0,0 +1,282 @@
+const ErrorResponse = require('../utils/errorResponse');
+const asyncHandler = require('../middleware/async');
+const LocationZone = require('../models/LocationZone');
+const Vehicle = require('../models/Vehicle');
+
+let INF = 10000;
+
+// @desc    Create a location zone
+// @route   POST /api/v1/locations/zones/
+// @access  Public
+exports.createLocationZone = asyncHandler(async (req, res, next) => {
+  const { path } = req.body;
+
+  // check that the path has a minimum of 3 vertices
+  if (path.length < 3) {
+    return next(
+      new ErrorResponse('A zone must have a defined perimeter with at least 3 points.', 400)
+    );
+  }
+
+  let locationBody = req.body;
+  // append the center to the body
+  locationBody.center = getPolyginCentroidAvg(path);
+
+  const locationZone = await LocationZone.create(locationBody);
+
+  res.status(201).json({
+    success: true,
+    payload: locationZone
+  })
+});
+
+// @desc    Get location zones for a dealership
+// @route   POST /api/v1/locations/zones/dealership/:dealershipId
+// @access  Public
+exports.getLocationZones = asyncHandler(async (req, res, next) => {
+  let locationZones = await LocationZone.find({ dealership: req.params.dealershipId });
+
+  for (let i = 0; i < locationZones.length; i++) {
+    let vehicleCount = await Vehicle.countDocuments({ zone: locationZones[i]._id });
+    locationZones[i].vehicleCount = vehicleCount;
+    locationZones[i].save();
+  }
```

Comment:
Using a for-of loop instead can make the code cleaner: 
`for (let zone of locationZones)`

Reply:
The for loop has been replaced by a for-of loop